assets/tests/TaskSchemaValidationServiceSpec.js

```diff
@@ -0,0 +1,301 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the QuestionSchemaValidatorService.
+ */
+
+describe('TaskSchemaValidationService', function() {
+  var QuestionDataService;
+  // TaskSchemaValidationService needs a shorter name.
+  var Tsvs;
+  var questions = [];
+  // Should contain all question IDs.
+  // TODO(eyurko): Figure out a way to dynamically check to make sure
+  // that all question IDs are specified.
+  var questionIds = ['reverseWords', 'parens', 'i18n', 'rle'];
+  globalData.questionSets.all = {};
+  globalData.questionSets.all.questionIds = questionIds;
+  globalData.questionSets.all.introductionParagraphs = [];
+  beforeEach(module('tie'));
+  beforeEach(inject(function($injector) {
+    // Used for testing the validator. Values will be inserted during the tests
+    // so that we don't have to redefine the dict every time.
+    QuestionDataService = $injector.get('QuestionDataService');
+    Tsvs = $injector.get('TaskSchemaValidationService');
+    QuestionDataService.initCurrentQuestionSet('all');
+    questions = [];
+    questionIds.forEach(function(item) {
+      questions.push(QuestionDataService.getQuestion(item));
+    });
+  }));
+
+  var assertTrueOnAllTasks = function(func, message) {
+    questions.forEach(function(item) {
+      var tasks = item.getTasks();
+      for (var i = 0; i < tasks.length; i++) {
+        if (message) {
+          expect(func(tasks[i])).toBe(true,
+            [item.getTitle(), message, i].join(''));
```

Comment:
You know -- I don't know that it's necessary to have the custom messages. You'd need to always keep them up to date, and it introduces rather strong coupling between the functions in Tsvs and the tests here. (I.e. if one changes then you need to remember to change the other.) On the flip side, if you just say the name of the function that failed in the error message, then the developer should probably be able to figure out what to fix.

What do you think about instead having a single test in this file that (for all tasks/questions) iterates on all methods in Tsvs and calls them? Then, to add a new check, you just add a new function to Tsvs and it "just works", and you don't have to repeat all the boilerplate of e.g. "at index ..." each time.